src/mame/f256/f256.cpp

```diff
@@ -0,0 +1,2008 @@
+// license:BSD-3-Clause
+// copyright-holders:Daniel Tremblay
+#include "emu.h"
+#include "f256.h"
+
+#include "cpu/m6502/w65c02.h"
+#include "tiny_vicky.h"
+
+/**
+ *
+ * F256K - WDC65C02 Processor running at 6.25MHz
+ *    512KB RAM managed with slots of 8KB in MMU located at address $0000.  Slots 0 to $3F
+ *    512KB Flash                                                           Slots $40 to $7F
+ *    All I/O are mapped to slot 6 ($C000-$DFFF) - there are 4 I/O maps, switched using address $0001.
+ *    Sound Chips: OPL3, PSG, SN74689, CODEC
+ *    Keyboard: mechanical switch in a matrix - controlled by VIA6522 chip.
+ *    Joysticks: 2 Atari-type ports and 2 S/NES ports
+ *    Mouse: over PS/2 - which can also be used for PS/2 Keyboard
+ *    IEC: Commodore Floppy Drive Controller
+ *
+ */
+
+f256_state::f256_state(const machine_config &mconfig, device_type type, const char *tag) :
+    driver_device(mconfig, type, tag)
+    , m_maincpu(*this, MAINCPU_TAG)
+    , m_ram(*this, RAM_TAG)
+    , m_iopage0(*this, IOPAGE0_TAG)
+    , m_iopage1(*this, IOPAGE1_TAG)
+    , m_iopage2(*this, IOPAGE2_TAG)
+    , m_iopage3(*this, IOPAGE3_TAG)
+    , m_rom(*this, ROM_TAG)
+    , m_font(*this, FONT_TAG)
+    , m_screen(*this, SCREEN_TAG)
+    , m_rtc(*this, "rtc")
+    , m_keyboard(*this, "ROW%u", 0)  // this, with the 8 array, requires 8 ROW of INPUTs
+    , m_via6522_0(*this, "via6522_0")
+	, m_via6522_1(*this, "via6522_1")
+
+    , m_sn0(*this, "sn76489_0")
+    , m_sn1(*this, "sn76489_1")
+    , m_opl3(*this, "ymf262")
+    , m_sid0(*this, "sid_0")
+    , m_sid1(*this, "sid_1")
+
+    , m_video(*this, "tiny_vicky")
+    //, m_iec(*this, "iec_bus")
+    //, m_iec_data_out(1)
+    //, m_iec_srq(1)
+    , m_ps2_keyboard(*this, "ps2_kbd")
+    , m_mouse(*this, "ps_mouse")
+    , m_uart(*this, "uart")
+    , m_sdcard(*this, "sdcard")
+	, m_spi_clock_state(false)
+	, m_spi_clock_sysclk(false)
+	, m_spi_clock_cycles(0)
+{
+}
+
+void f256_state::f256k(machine_config &config)
+{
+    W65C02(config, m_maincpu, MASTER_CLOCK/4);
```

Comment:
I believe this should be using the W65C02S variant.

Reply:
OK. I've updated this.